### PR TITLE
Mid task number 2: "User Cleanup"

### PR DESCRIPTION
**Pull request by sernamar**

### CODE REVIEW

File: stats_tracker/stats_tracker/leaderboards/leaderboards.py [hunk 1]
```diff
@@ -147,3 +147,20 @@ def _get_leaderboard_set_id(product_id: str, sub_key: str) -> str:
...
+        key_pattern = _get_leaderboard_set_id("*", "wins")
+        redis_db = get_stats_tracker_db()
+        leaderboard_keys = redis_db.keys(key_pattern)
+        for key in leaderboard_keys:
+            logging.debug(f"Deleting {user_id} from {key}.")
+            num_users_deleted = redis_db.zrem(key, user_id)
+            if num_users_deleted > 0:
+                leaderboards_affected.append(key.decode())
+        return leaderboards_affected
+    except:
```
Comment (Code-Over-Time) [inline stats_tracker/stats_tracker/leaderboards/leaderboards.py:165]: It's worth logging some information about the exception here. If there are errors happening in a live system you want to know exactly what happened. You can still return none, which implies and error happened - but you will want to leave a paper trail

File: stats_tracker/tests/leaderboards/test_leaderboards.py [hunk 2]
```diff
@@ -71,6 +71,30 @@ def test_get_leaderboard(self, stats_tracker_db, test_client):
...
+
+        # 4. Test that "userA" does not exist in the leaderboard
+        response = test_client.get(f"leaderboards/top_ten/exrps")
+        assert response.status_code == 200
+        assert not "userA" in json.loads(response.data)[0].values()
+
+        # 5. Delete again "userA" from the leaderboard (now "userA" does not exist in the leaderboard)
+        response = test_client.delete(f"leaderboards/user/userA")
+        assert response.status_code == 200
+        assert json.loads(response.data) == []
```
Comment (Code-Over-Time) [inline stats_tracker/tests/leaderboards/test_leaderboards.py:96]: It's also worth adding a unit test for the delete_user function in leaderboards.py. This test covers the functionality from a high level, but you haven't covered the failure scenario (ie. can't access redis). You always want to make sure you don't simply cover the 'happy path'. You want to test to make sure the system handles bad inputs and system outages as expected.

Here is a great talk on this subject: https://www.youtube.com/watch?v=IiGXq3yY70o

File: website/server/tests/game_portal/accounts/test_accounts.py [hunk 3]
```diff
@@ -70,3 +70,16 @@ def test_create_user_with_invalid_email(self, api_client):
...
+    def test_signal_delete_user_from_leaderboards_triggered(self, mock_delete_user, mock_requests):
+        # Create and delete a user to trigger the delete_user_from_leaderboards signal
+        new_user = Account.objects.create_user("userA", "userA@test.com", "validPassword")
+        new_user.delete()
+
+        # Check that the signal was called
+        assert mock_delete_user.called
+
+        # Check that the signal was called only once
+        assert mock_delete_user.call_count == 1
```
Comment (Code-Over-Time) [inline website/server/tests/game_portal/accounts/test_accounts.py:85]: Similar comment here - this test covers the happy path, but you should add coverage for when the request to delete user fails - as this is quite a realistic scenario (eg. if the stats service is under load or something).

File: stats_tracker/stats_tracker/leaderboards/leaderboards.py (hunk 1 above)
Comment (sernamar) [reply stats_tracker/stats_tracker/leaderboards/leaderboards.py]: Fixed in [940099c](https://github.com/sernamar/varcade_games/pull/9/commits/940099c7146e3036322c82a20615cd3ed00be1a6)

File: stats_tracker/tests/leaderboards/test_leaderboards.py (hunk 2 above)
Comment (sernamar) [reply stats_tracker/tests/leaderboards/test_leaderboards.py:130]: Fixed in [d10e22d](https://github.com/sernamar/varcade_games/pull/9/commits/d10e22d17459fb91bb2d9f6f9b0c83ad6c2be0bd)

File: website/server/tests/game_portal/accounts/test_accounts.py (hunk 3 above)
Comment (sernamar) [reply website/server/tests/game_portal/accounts/test_accounts.py:87]: Fixed in [4fb8c53](https://github.com/sernamar/varcade_games/pull/9/commits/4fb8c53886cf608fdd560c4f59aa1b077f3f3301)

**Pull request merged**